lib/maze/servlets/servlet.rb

```diff
@@ -3,8 +3,27 @@
 require 'zlib'
 require 'stringio'
 require 'json_schemer'
+require 'delegate'
 
 module Maze
+  class HttpRequest < SimpleDelegator
+    def body
+      @body ||= decode_body
+    end
+
+    private
+
+    def decode_body
+      delegate = __getobj__
+      if %r{^gzip$}.match(delegate['Content-Encoding'])
+        gz_element = Zlib::GzipReader.new(StringIO.new(delegate.body))
+        gz_element.read
+      else
+        delegate.body
+      end
+    end
+  end
```

Comment:
I'd prefer this was in its own file, `http_request.rb` - it'll make its existence more obvious and I think there's a good chance that we could wrap more functionality into it.

Reply:
If you're happy to move it, we'll also need a new `require_relative` line in `bin/maze-runner`.